script/generate_prediction_in_large_batches.py

```diff
@@ -0,0 +1,135 @@
+import json
+from tqdm import tqdm
+import sys
+
+sys.path.append('./')
+from helper.global_var import FLAG_DEBUG, SAVE_TYPE_PICKLE, PREDICT_ROAD_CONDITION_CONFIG_HISTORY_DATE, \
+    PREDICT_ROAD_CONDITION_CONFIG_HISTORY_DATA_RANGE, PREDICT_ROAD_CONDITION_CONFIG_WEIGHT
+from pathlib import Path
+from helper.helper_time_range_index_to_str import time_range_index_to_str, time_range_index_to_time_range_str
+import predict_road_condition
+from helper.global_var import SAVE_TYPE_PICKLE
+from helper.graph_reader import graph_reader
+from datetime import datetime, timedelta
+
+
+def predict_speed_dict_to_json(predict_speed_dict, target_date_str, time_slot_interval, interval_idx,
+                               final_way_table, way_types, way_type_avg_speed_limit,
+                               save_path="cache/predict_result/{0}/{1}/{2}.json"):
```

Comment:
Why do you use this format in parameters? Can we just pass in the formatted (final) string? This happens in a few of other places as well.